### ISSUE TITLE
Model file not found: assets/hubert/hubert_base.pt

### ISSUE THREAD

**Issue by kevin-guimard-ext** [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#1445]
Hello,

I ran the GUI and, in inference mode, selected an index file of a pretrained model and the path of the audio file to be processed. But I run into the following error:
```
  File "/home/jupyter/Retrieval-based-Voice-Conversion-WebUI/infer/modules/vc/modules.py", line 170, in vc_single
    self.hubert_model = load_hubert(self.config)
  File "/home/jupyter/Retrieval-based-Voice-Conversion-WebUI/infer/modules/vc/utils.py", line 23, in load_hubert
    models, _, _ = checkpoint_utils.load_model_ensemble_and_task(
  File "/opt/conda/lib/python3.10/site-packages/fairseq/checkpoint_utils.py", line 423, in load_model_ensemble_and_task
    raise IOError("Model file not found: {}".format(filename))
OSError: Model file not found: assets/hubert/hubert_base.pt
```

Is Hubert the base model that this app will use? Don't I have to input a pth file also?

Thanks,
Kevin

**Comment by Misaka-Mikodo** [comment]
I had the same problem.

**Comment by marshalleq** [comment]
And did you manage to fix it?

**Comment by Access-Rend** [comment]
+1 hope a solution

**Comment by faustofonseca** [comment]
I have the same exact issue on my mac.

**Comment by Tessory** [comment]
Yes, it's strange bug. Sometimes I also saw that error, in that case I choose one standard model and convert one mp3 file, after that choose back custom model and it begin work fine without error... until I restart RVC server

**Comment by PJaekel93** [comment]
any news on this? Bug still persist, having the same issue. Tried a lot of things without success.

**Comment by Tr-AIL** [comment]
I guess it's because the program cannot parse non-windows file paths. I checked similar error analysis articles, which mentioned this. I encountered the same error under Linux, but everything was normal under Windows.

**Comment by faustofonseca** [comment]
@Tr-AIL thats an interesting point. But would that mean that no one actually made this work on Mac or Windows? I always assumed since the tutorial talks about Mac.  Or do you mean the paths that we provide in the UI?

**Comment by PJaekel93** [comment]
thanks for replying. I managed to find a solution for myself and got it running, it might have had multiple reasons. The error originates from a problem where an asset is not correctly loaded.

1. Wrong file paths: indeed in one of the earlier versions the filepath was wrong. It was fixed in a later update and should work now. 
2. Fault on my side, wrong use of the program. I did not run the full train, so the dropdown to chose a voice was empty. If any asset that is needed is empty it might throw the above mentioned error.

**Comment by faustofonseca** [comment]
Oh, nice. Thank you for sharing!

**Comment by artemious3** [comment]
Make sure that the .pth model file is placed into `assets/weights/<NAME>.pth`, and .index model file is `logs/<NAME>/added_<...>.index`
Notice that .pth file name and folder name in logs are the same.

**Comment by sankexin** [comment]
solution：

![image](https://github.com/user-attachments/assets/8c02bc9e-da02-4f3f-9b78-aa7805f4e657)

![image](https://github.com/user-attachments/assets/650842a1-0b7e-4c97-8b28-cd2379d504f7)



2024-07-19 17:06:03 | INFO | infer.modules.vc.modules | Loading: assets/weights/mi-test.pth
2024-07-19 17:06:04 | INFO | infer.modules.vc.modules | Select index: logs/mi-test/added_IVF1043_Flat_nprobe_1_mi-test_v2.index
2024-07-19 17:06:44 | INFO | infer.modules.vc.pipeline | Loading rmvpe model,assets/rmvpe/rmvpe.pt

**Comment by chuksum** [comment]
Reinstalling the exact packages in the requirements.txt fixed it on my end :

`pip install -r requirements.txt`

Some versions had mismatches and caused these weird types of incompatibility glitches.

**Comment by wayne931121** [comment]
Hello, today I meet this problem, and it is my solution:

1. see error code:
```
  File "C:\ai\lib\site-packages\gradio\queueing.py", line 745, in process_events
    response = await route_utils.call_process_api(
  File "C:\ai\lib\site-packages\gradio\route_utils.py", line 354, in call_process_api
    output = await app.get_blocks().process_api(
  File "C:\ai\lib\site-packages\gradio\blocks.py", line 2127, in process_api
    data = await self.postprocess_data(block_fn, result["prediction"], state)
  File "C:\ai\lib\site-packages\gradio\blocks.py", line 1904, in postprocess_data
    prediction_value = block.postprocess(prediction_value)
  File "C:\ai\lib\site-packages\gradio\components\audio.py", line 316, in postprocess
    file_path = processing_utils.save_audio_to_cache(
  File "C:\ai\lib\site-packages\gradio\processing_utils.py", line 182, in save_audio_to_cache
    temp_dir = Path(cache_dir) / hash_bytes(data.tobytes())
AttributeError: 'NoneType' object has no attribute 'tobytes'
```

2. you can found that it is the gradio problem

3. reinstall correct gradio, i install gradio==3.34.0 in
https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/main/requirements.txt

4. find more problem:
 (maybe you will not meet this problem, my some package version is different with requirements.txt, because i want to use this env for other projects, so please check the solution above, if not work, continue to below.)
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for PredictBody event_id Field required [type=missing, input_value={'data': ['describe this ...on_hash': 'ncvk1jrssin'}, input_type=dict] 
```

5. search in internet, get result
https://github.com/gradio-app/gradio/issues/6914

6. run the code
```
pip install gradio==3.50.2
```

7. success fix this

**Comment by Leon-LihongWang** [comment]
> solution：
> 
> ![image](https://github.com/user-attachments/assets/8c02bc9e-da02-4f3f-9b78-aa7805f4e657)
> 
> ![image](https://github.com/user-attachments/assets/650842a1-0b7e-4c97-8b28-cd2379d504f7)
> 
> 2024-07-19 17:06:03 | INFO | infer.modules.vc.modules | Loading: assets/weights/mi-test.pth 2024-07-19 17:06:04 | INFO | infer.modules.vc.modules | Select index: logs/mi-test/added_IVF1043_Flat_nprobe_1_mi-test_v2.index 2024-07-19 17:06:44 | INFO | infer.modules.vc.pipeline | Loading rmvpe model,assets/rmvpe/rmvpe.pt

This approach is very close to the correct solution. I noticed that if I **open a fresh browser window** after running `python infer-web.py` and perform the operation there, the error no longer occurs. However, if I reuse a previously opened browser tab/window (with parameters already filled in) and click the button again, the error is triggered.

**Comment by sankexin** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
